packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js

```diff
@@ -96,6 +96,27 @@ export function useFocusFirstElement( clientId ) {
 			return;
 		}
 
+		// Check to see if Block contains focussable element before a generic caret insert.
+		if ( ! ref.current.getAttribute( 'contenteditable' ) ) {
+			let focusElement = focus.tabbable.findNext( ref.current );
+			// Need to make sure we're still in the current Block, if not, we could run in to trouble if focus is placed in a Block further down the page.
+			if ( ! ref.current.contains( focusElement ) ) {
+				return;
+			}
+			// Need to try our best to detect child Blocks and not focus them straight away.
+			const disallowedTags = [ 'div', 'figure' ];
+			const checkDisallowedTags = disallowedTags.some(
+				( tag ) => focusElement.tagName.toLowerCase() === tag
+			);
+			if ( checkDisallowedTags ) {
+				focusElement = false;
+			}
+			if ( focusElement ) {
+				focusElement.focus();
+				return;
+			}
+		}
+
 		setContentEditableWrapper( ref.current, false );
```

Comment:
@ellatrix I am not sure what this line is for right off hand, is it okay to have my code run before this or do I need to switch the order?